bin/ci/self-test/script.sh

```diff
@@ -15,13 +15,13 @@ cd ../../../ || exit 1
 
 XDEBUG_IS_ENABLED=$(php -r 'echo function_exists("xdebug_get_code_coverage") ? "TRUE" : "FALSE";')
 
-if [[ "$ORCA_COVERAGE_ENABLE" == TRUE && "$XDEBUG_IS_ENABLED" == "FALSE" ]]; then
-  echo "ORCA_COVERAGE_ENABLE is on but Xdebug is disabled"
+if [[ "$ORCA_COVERAGE_ENABLE" == TRUE  || "$ORCA_COVERAGE_COBERTURA_ENABLE" == TRUE  || "$ORCA_COVERAGE_CLOVER_ENABLE" == TRUE ]] && [ "$XDEBUG_IS_ENABLED" == FALSE ]; then
+  echo "Coverage generation is on but Xdebug is disabled"
   exit 1
 fi
 
-if [[ "$ORCA_COVERAGE_ENABLE" == FALSE && "$XDEBUG_IS_ENABLED" == "TRUE" ]]; then
-  echo "ORCA_COVERAGE_ENABLE is off but Xdebug is enabled"
+if [[ "$ORCA_COVERAGE_ENABLE" == FALSE  && "$ORCA_COVERAGE_COBERTURA_ENABLE" == FALSE  && "$ORCA_COVERAGE_CLOVER_ENABLE" == FALSE ]] && [ "$XDEBUG_IS_ENABLED" == TRUE ]; then
+  echo "Coverage generation is off but Xdebug is enabled"
   exit 1
 fi
```

Comment:
These conditions are complex and hard to read. They suggest a need for refactoring and simplification. What if somewhere in the includes we automatically set `$ORCA_COVERAGE_ENABLE` to `TRUE` if _either_ `$ORCA_COVERAGE_COBERTURA_ENABLE` or `$ORCA_COVERAGE_CLOVER_ENABLE` is on and then use it under the hood for simpler comparisons. In fact, if we did that, I don't think these conditional tests would have to change at all. (Later, since we're deprecating `$ORCA_COVERAGE_ENABLE`, we can continue to use it under the hood and just not allow users to override it.)

Reply:
I think it would create a conflict if we automatically set $`ORCA_COVERAGE_ENABLE` to `TRUE` when `ORCA_COVERAGE_COBERTURA_ENABLE` is _ON_ because `ORCA_COVERAGE_ENABLE` is an alias of `$ORCA_COVERAGE_CLOVER_ENABLE`.

Reply:
Okay, that makes sense. Well, readability is my only real concern here, so what if we just introduce a temporary variable. Something like this, maybe? (My bash is a little rusty.)

```shell
if [[ "$ORCA_COVERAGE_ENABLE" == TRUE  || "$ORCA_COVERAGE_COBERTURA_ENABLE" == TRUE  || "$ORCA_COVERAGE_CLOVER_ENABLE" == TRUE ]]; then
  ANY_COVERAGE_IS_ENABLED=TRUE
else
  ANY_COVERAGE_IS_ENABLED=TRUE
fi
```

Reply:
I have created this` ORCA_ANY_COVERAGE_IS_ENABLED` and neither added it to any documentation and nor added it to services to keep this under the hood.